### PR TITLE
add Dockerfiles and Makefile, test cases and README for docker contai…

### PR DESCRIPTION
**Pull request by ScottABrown**
Run "make help" for documentation.

This adds Dockerfiles to create a base image from which docker images for the api server and data collector are built. It will also build and initialize a shared sqlite data volume if specified.

It would be good to try this out in more environments; I've tested it with the test file and manually and it looks good, but I didn't test it with swag or with a non-sqlite data URI.

I also need an example of a non-sqlite database URI for the documentation.

### CODE REVIEW

File: docker/Dockerfile.aardvark-base [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+LABEL author="scottbrown0001@gmail.com"
+
+COPY phantomjs/bin/phantomjs /usr/local/bin
+
+COPY aardvark /tmp/aardvark
+WORKDIR /tmp/aardvark
+
+# # Update the repository and install tools and aardvark.
+# RUN apt-get update && \
+# 	apt-get install -y vim && \
```
Comment (mcpeak) [inline docker/Dockerfile.aardvark-base]: do we need vim?

Comment (ScottABrown) [reply docker/Dockerfile.aardvark-base]: Good catch. Not needed for anything in the build-run path. I also see I left the update step commented out by mistake - doubly good catch. Will commit an update in a sec.

Comment (ScottABrown) [reply docker/Dockerfile.aardvark-base]: Ok, that was a long sec. Re-ran the test file and found an issue with how the AARDVARK_IMAGES_TAG env variable was handled, fixed that too and took the time to clean up some highly repetitive code, for posterity.

**Comment by mcpeak** [comment]
Awesome work!
\

**Pull request merged**